binaries/build.py

```diff
@@ -1,44 +1,78 @@
 import argparse
+import glob
 import os
 import sys
-import glob
 
 # To help discover local modules
 REPO_ROOT = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..")
 sys.path.append(REPO_ROOT)
 
-from pathlib import Path
 
-from ts_scripts.utils import is_conda_env, is_conda_build_env
-from binaries.conda.build_packages import conda_build, install_miniconda, install_conda_build
+from binaries.conda.build_packages import (
+    conda_build,
+    install_conda_build,
+    install_miniconda,
+)
+from ts_scripts.utils import is_conda_build_env, is_conda_env
 
 
-def build():
-    
-    print("## Started torchserve, model-archiver and workflow-archiver build")
-    create_wheel_cmd = "python setup.py bdist_wheel --release --universal"
+def build_dist_whl(nightly=False):
+    """
+    Function to build the wheel files for torchserve, model-archiver and workflow-archiver
+    nightly is an optional argument to build nightlies
+    """
 
-    # Build torchserve wheel
-    print(f"## In directory: {os.getcwd()} | Executing command: {create_wheel_cmd}")
-    ts_build_exit_code = os.system(create_wheel_cmd)
+    binaries = ["torchserve", "torch-model-archiver", "torch-workflow-archiver"]
+    if nightly:
+        print(
+            "## Started torchserve, model-archiver and workflow-archiver nightly build"
+        )
+        create_wheel_cmd = "python setup.py "
+    else:
+        print("## Started torchserve, model-archiver and workflow-archiver build")
+        create_wheel_cmd = "python setup.py bdist_wheel --release --universal"
 
-    # Build model archiver wheel
-    os.chdir("model-archiver")
-    print(f"## In directory: {os.getcwd()} | Executing command: {create_wheel_cmd}")
-    ma_build_exit_code = os.system(create_wheel_cmd)
+    for binary in binaries:
+        if "serve" in binary:
+            cur_dir = REPO_ROOT
+        else:
+            cur_dir = REPO_ROOT + "/" + binary[len("torch-") :]
```

Comment:
This will break on Windows - use `os.path.join` instead

Reply:
Done